low-level/framework/utils/ipmi_client.py

```diff
@@ -252,6 +252,47 @@ def _run_ipmitool_subcommand(self, subcommand, grep_args=None):
 
         return out, error, retcode
 
+    def get_server_fru_list(self):
+        """Get FRU info from server and load it into Config."""
+        try:
+            server_frus = Conf.get(SSPL_CONF, "SYSTEM_INFORMATION>server_fru_list")
+        except ValueError as e:
+            logger.error("Failed to get server_fru_list from config."
+                         f"Error:{e}")
+        cmd = 'fru list'
+        fru_list = []
+        out, err, ret = self._run_ipmitool_subcommand(cmd,
+        grep_args="FRU Device Description : ")
+        if ret != 0:
+            logger.error("Failed in fetching FRU info from server."
+                         f"Error:{err}")
+        if out:
+            for l in out.split('\n'):
+                fru_list.append(l.split(': ')[1])
+            keywords = ['Pwr Supply', 'power', 'PS', 'PSU']
```

Comment:
PSU is represented in such different manner in different servers ?

Reply:
yes..
on intel server
>FRU Device Description : **Pwr Supply 1 FRU (ID 2)**
 Product Manufacturer  : FLEXTRONICS
 Product Name          : S-1100ADU00-201
 Product Part Number   : G84027-010
 Product Version       : 04
 Product Serial        : EXWD90700475

FRU Device Description : Pwr Supply 2 FRU (ID 3)
 Product Manufacturer  : FLEXTRONICS
 Product Name          : S-1100ADU00-201
 Product Part Number   : G84027-010
 Product Version       : 04
 Product Serial        : EXWD90700623

on dell server:

> FRU Device Description : **PS1 (ID 1)**
 Board Mfg Date        : Sat Jun 19 18:00:00 2010
 Board Mfg             : DELL
 Board Product         : PWR SPLY,750W,RDNT,LTON
 Board Serial          : CNLOD0006K2ADD
 Board Part Number     : 0KTW3MA00

FRU Device Description : PS2 (ID 2)
 Board Mfg Date        : Sat Jun 19 18:00:00 2010
 Board Mfg             : DELL
 Board Product         : PWR SPLY,750W,RDNT,LTON
 Board Serial          : CNLOD0006K4E34
 Board Part Number     : 0KTW3MA00